lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt

```diff
@@ -28,48 +41,161 @@ class FindMealFragment : Fragment() {
         mealsRecyclerView.adapter = mealsAdapter
         mealsRecyclerView.layoutManager = LinearLayoutManager(context)
 
-        swipeRefreshLayout = view.findViewById(R.id.swipeContainer)
         swipeRefreshLayout.setOnRefreshListener {
             swipeRefreshLayout.isRefreshing = false
-            displayMeals(true)
+            getLastLocation()
         }
 
-        displayMeals(false)
+        getLastLocation()
+    }
 
-        return view
+    private fun confirmPermissions(context: Context) {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) !== PackageManager.PERMISSION_GRANTED) {
+            requestFineLocationPermission(LOCATION_PERMISSION_REQUEST_CODE)
+        }
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel = ViewModelProvider(this).get(FindMealViewModel::class.java)
-        // TODO: Use the ViewModel
+    fun requestFineLocationPermission(requestCode: Int = 0) {
+        requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), requestCode)
     }
 
-    private fun displayMeals(needReload: Boolean) {
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        if (permissionResultParser.isFineLocationPermissionsGranted(permissions, grantResults)) {
+            getLastLocation()
+        } else {
+            // TODO Ask user to update settings / location settings are required
+        }
+    }
+
+    fun getLastLocation() {
+        fusedLocationClient.lastLocation.addOnSuccessListener { location: Location ->
+            lastLocation = location
+            updateMeals(false)
+        }
+    }
+
+    private fun startLocationUpdates() {
+        val locationRequest = LocationRequest()
+
+        locationRequest.interval = 100000
+        locationRequest.fastestInterval = 5000
+        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+
+        if (!hasSetupLocationUpdates) {
+            setupLocationUpdates(locationRequest)
+        }
+        fusedLocationClient.requestLocationUpdates(locationRequest, locationCallback, null)
+    }
+
+    private fun setupLocationUpdates(locationRequest: LocationRequest) {
+        val builder = LocationSettingsRequest.Builder()
+            .addLocationRequest(locationRequest)
+        val client = LocationServices.getSettingsClient(this.requireActivity())
+        val task = client.checkLocationSettings(builder.build())
+
+        task.addOnSuccessListener {
+            locationUpdateState = true
+        }
+        task.addOnFailureListener { e ->
+           if (e is ResolvableApiException) {
+               try {
+                   e.startResolutionForResult(this.requireActivity(), REQUEST_CHECK_SETTINGS)
+               } catch (sendEx: IntentSender.SendIntentException) {
+                   //
```

Comment:
something missing?

Reply:
Yes, have updated the catch now to Log exception out